### ISSUE TITLE
Cannot build driver

### ISSUE THREAD

**Issue by gaberilde** [tomaspinho/rtl8821ce#58]
The error code is this
`/var/lib/dkms/rtl8821ce/v5.5.2_34066.20190614/build/os_dep/linux/rtw_android.c: In function ‘rtw_android_priv_cmd’:
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20190614/build/os_dep/linux/rtw_android.c:665:49: error: macro "access_ok" requires 3 arguments, but only 2 given
  if (!access_ok(priv_cmd.buf, priv_cmd.total_len)) {
                                                 ^
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20190614/build/os_dep/linux/rtw_android.c:665:7: error: ‘access_ok’ undeclared (first use in this function)
  if (!access_ok(priv_cmd.buf, priv_cmd.total_len)) {
       ^~~~~~~~~
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20190614/build/os_dep/linux/rtw_android.c:665:7: note: each undeclared identifier is reported only once for each function it appears in
/usr/src/linux-headers-4.9.0-8-common/scripts/Makefile.build:309: recipe for target '/var/lib/dkms/rtl8821ce/v5.5.2_34066.20190614/build/os_dep/linux/rtw_android.o' failed
make[4]: *** [/var/lib/dkms/rtl8821ce/v5.5.2_34066.20190614/build/os_dep/linux/rtw_android.o] Error 1
make[4]: *** Waiting for unfinished jobs....
/usr/src/linux-headers-4.9.0-8-common/Makefile:1546: recipe for target '_module_/var/lib/dkms/rtl8821ce/v5.5.2_34066.20190614/build' failed
make[3]: *** [_module_/var/lib/dkms/rtl8821ce/v5.5.2_34066.20190614/build] Error 2
Makefile:152: recipe for target 'sub-make' failed
make[2]: *** [sub-make] Error 2
Makefile:8: recipe for target 'all' failed
make[1]: *** [all] Error 2
make[1]: Leaving directory '/usr/src/linux-headers-4.9.0-8-amd64'
Makefile:2237: recipe for target 'modules' failed
make: *** [modules] Error 2
`

**Comment by gaberilde** [comment]
The kernal version is 4.9.0-8-amd64

**Comment by d1mertz1** [comment]
I had the same issue with a brand new hp laptop and linux mint 19.2

As a workaround, building from an older revision of the repository worked fine.
I used the 62742760ce5029e52c197632d641fe5c40a06db7 from 18 Jul, which is the last one before new source code from upstream was pulled.

Just do a 
``git checkout 62742760ce5029e52c197632d641fe5c40a06db7

after the git clone 

Best
Denis

**Comment by gaberilde** [comment]
yeah i downloaded another one but it killed the os so we install xubuntu instead and this one worked  then.

**Comment by hackson99** [comment]
> The error code is this
> `/var/lib/dkms/rtl8821ce/v5.5.2_34066.20190614/build/os_dep/linux/rtw_android.c: In function ‘rtw_android_priv_cmd’: /var/lib/dkms/rtl8821ce/v5.5.2_34066.20190614/build/os_dep/linux/rtw_android.c:665:49: error: macro "access_ok" requires 3 arguments, but only 2 given if (!access_ok(priv_cmd.buf, priv_cmd.total_len)) { ^ /var/lib/dkms/rtl8821ce/v5.5.2_34066.20190614/build/os_dep/linux/rtw_android.c:665:7: error: ‘access_ok’ undeclared (first use in this function) if (!access_ok(priv_cmd.buf, priv_cmd.total_len)) { ^~~~~~~~~ /var/lib/dkms/rtl8821ce/v5.5.2_34066.20190614/build/os_dep/linux/rtw_android.c:665:7: note: each undeclared identifier is reported only once for each function it appears in /usr/src/linux-headers-4.9.0-8-common/scripts/Makefile.build:309: recipe for target '/var/lib/dkms/rtl8821ce/v5.5.2_34066.20190614/build/os_dep/linux/rtw_android.o' failed make[4]: *** [/var/lib/dkms/rtl8821ce/v5.5.2_34066.20190614/build/os_dep/linux/rtw_android.o] Error 1 make[4]: *** Waiting for unfinished jobs.... /usr/src/linux-headers-4.9.0-8-common/Makefile:1546: recipe for target '_module_/var/lib/dkms/rtl8821ce/v5.5.2_34066.20190614/build' failed make[3]: *** [_module_/var/lib/dkms/rtl8821ce/v5.5.2_34066.20190614/build] Error 2 Makefile:152: recipe for target 'sub-make' failed make[2]: *** [sub-make] Error 2 Makefile:8: recipe for target 'all' failed make[1]: *** [all] Error 2 make[1]: Leaving directory '/usr/src/linux-headers-4.9.0-8-amd64' Makefile:2237: recipe for target 'modules' failed make: *** [modules] Error 2 `

Hi, i got the same error info on my debian10. How do you resolve the problem?
Hope you reply.
Thank u.

**Comment by gaberilde** [comment]
You have to use another driver version (see https://github.com/realtekofflinedrivers/rtl8821ce-offlineinstaller/tree/master/driver3/Debian)
Or do as the other reply said to download this and commit a old version that dosnet have this issue.

**Comment by tomaspinho** [comment]
Duplicate of https://github.com/tomaspinho/rtl8821ce/issues/57
